src/core/app_config/index.ts

```diff
@@ -52,8 +69,42 @@ export default (() => {
         url: 'https://www.openstreetmap.org/edit?editor=remote#map=',
       },
     ],
+    effectiveFeatures: {},
   };
-})();
+  configs['global'] = globalAppConfig;
+  return globalAppConfig;
+}
+
+function getEffectiveFeatures(appConfig: AppConfigEffective) {
+  return Object.fromEntries(
+    [...(appConfig.featuresByDefault ?? []), ...(appConfig.features ?? [])].map((k) => [
+      k,
+      true,
+    ]),
+  );
+}
+
+export function getEffectiveConfig(): AppConfigEffective {
+  getGlobalConfig();
+  const mergedAppConfig = { ...configs.global, ...configs.custom };
+  mergedAppConfig.effectiveFeatures = getEffectiveFeatures(mergedAppConfig);
+  // mutate object to keep reference for export
+  Object.assign(configs.merged, mergedAppConfig);
```

Comment:
wouldn't `configs.merged = mergedAppConfig` keep the reference and mutate object too?

Reply:
No